src/main/kotlin/com/virtuslab/dokka/site/StaticSiteContext.kt

```diff
@@ -141,7 +134,59 @@ class StaticSiteContext(val root: File, cxt: DokkaContext) {
                 content
             )
         }
-        return all.map { templateToPage(it) }
+        return all.map(::templateToPage)
     }
-}
 
+    private fun getExternalDriResolver(dri: DRI, allDRIs: Map<String, DRI>): (String) -> DRI? = {
+        if (it.endsWith(".html") || it.endsWith(".md")) {
+            it.resolveLinkToFile(dri, allDRIs)
+        } else {
+            it.replace("\\s".toRegex(), "").resolveLinkToApi()
+        }
+    }
+
+    private fun String.resolveLinkToFile(dri: DRI, allDRIs: Map<String, DRI>) =
+        if (startsWith("/")) { // handle root related links
+            replace('/', '.').removePrefix(".")
+
+        } else { // handle relative links
+            val unSuffixedDri = dri.packageName!!.removeSuffix(".html").removeSuffix(".md")
+            val parentDri = unSuffixedDri.take(unSuffixedDri.indexOfLast('.'::equals)).removePrefix("_.")
+            "${parentDri}.${replace('/', '.')}"
+
+        }.let(allDRIs::get)
+
+    private fun String.resolveLinkToApi() = when {
```

Comment:
This will work only partially for `Kotlin` since DRIs are a bir more complicated. @Kordyjan @kamildoleglo could you help here? Maybe we can reuse dokka code?

We should also create extension point to that will alow users to provide its own mapping String -> DRI?

Reply:
TBH I'd personally discourage creating `DRIs` by hand if possible. Reusing the IntelliJ `resolveKDocLink` method feels more correct here. It will certainly limit the types of links possible (eg. no linking to overloaded methods, because the KDoc specification does not allow it for now), but will be more fail-proof. 

If you have to create the `DRIs` manually though, the `JavaClassReference` looks wrong for starters (not really sure what issue it resolves in Dokka TBH, I'll try to find out). Moreover, toplevel methods (classes, etc.) should have a `null` `packageName`, not a blank one. Also looks like it's going to fail for elements with generic attributes, because you're not creating `TypeReferences` anywhere and for extension methods (and properties), because you're not setting the `receiver` anywhere